src/apps/api/serializers/queues.py

```diff
@@ -30,28 +32,33 @@ class Meta:
             'created_when',
             'is_owner',
             'id',
-        ]
-        read_only_fields = [
+        )
+        read_only_fields = (
             'broker_url',
             'vhost',
             'created_when',
             'is_owner',
-        ]
+        )
+
+    def validate(self, attrs):
+        request = self.context.get('request')
+        if request.user.queues.count() == request.user.rabbitmq_queue_limit and not request.user.is_superuser:
+            # TODO: Do we want to raise a validation error (400, bad request) or something that will raise a 403?
+            raise ValidationError("User has reached queue limit!")
```

Comment:
Talked to Eric, we think we should probably raise `PermissionDenied` instead of `ValidationError` here, so the correct error will get sent back to the front end.